tests/e2e_tests/integration/07-saml.spec.ts

```diff
@@ -0,0 +1,246 @@
+      await page.screenshot({path: 'saml-uploaded.png'});
+    });
+
+    // Creates a user with login that matches Identity privder (samltest.id) user email
+    test('Creates a user without a password', async ({ environment, samlCredentials, context, baseUrl, page }) => {
+      test.skip(environment !== 'staging');
+      const response = await axios({
+        url: `${baseUrl}api/management/v1/useradm/users?email=${encodeURIComponent(samlSettings['credentials'][samlCredentials]['email'])}`,
+        method: 'GET',
+        headers: {
+          Authorization: `Bearer ${jwt}`,
+          'Content-Type': 'application/json'
+        },
+        httpsAgent: new https.Agent({
+          rejectUnauthorized: false
+        })
+      });
+
+      const responseStatus = await response.status;
+      if (responseStatus != 200 || response.data.length < 1) {
+        await page.goto(`${baseUrl}ui/settings/user-management`);
+        // Click text=Create new user
+        await page.locator('text=Create new user').click();
+        // Click [placeholder="Email"]
+        await page.locator('[placeholder="Email"]').click();
+        // Fill [placeholder="Email"]
+        await page.locator('[placeholder="Email"]').fill(samlSettings['credentials'][samlCredentials]['email']);
+        // Click text=Create user
+        await page.locator('text=Create user').click();
+        await page.waitForTimeout(timeoutOneSecond);
+        await expect(page.locator(`text=${samlSettings['credentials'][samlCredentials]['email']}`)).toHaveText(samlSettings['credentials'][samlCredentials]['email']);
+        console.log(`${samlSettings['credentials'][samlCredentials]['email']} created.`);
+      } else {
+        console.log(`${samlSettings['credentials'][samlCredentials]['email']} already exists.`);
+      }
+    });
+
+    // This test calls auth/sso/${id}/login, where id is the id of the identity provider
+    // and verifies that login is successful.
+    test('User can login via sso/login endpoint', async ({environment, samlCredentials, baseUrl, browser}) => {
+      test.skip(environment !== 'staging');
+      console.log(`logging in via ${baseUrl}api/management/v1/useradm/auth/sso/${metadataId}/login (credentials set:${samlCredentials})`);
+      const context = await browser.newContext()
+      const page = await context.newPage()
+      await page.goto(`${baseUrl}api/management/v1/useradm/auth/sso/${metadataId}/login`);
+      await page.waitForTimeout(timeoutFourSeconds);
+      // This screenshot saves the view right after the first redirection
+      await page.screenshot({path: 'saml-redirected.png'});
+
+      // Click input[name="j_username"]
+      await page.locator('input[name="j_username"]').click();
+
+      // Fill input[name="j_username"]
+      await page.locator('input[name="j_username"]').fill(samlSettings['credentials'][samlCredentials]['login']);
+
+      // Press Tab
+      await page.locator('input[name="j_username"]').press('Tab');
```

Comment:
unless this is needed, please remove it, as it only increases reliance on the page layout.

Reply:
removed. nice catch, I thought I removed all of them.